app/src/main/java/org/p2p/wallet/history/interactor/mapper/RpcHistoryTransactionConverter.kt

```diff
@@ -77,6 +79,10 @@ class RpcHistoryTransactionConverter(
     private fun parseSwap(transaction: RpcHistoryTransactionResponse): RpcHistoryTransaction {
         val info = gson.fromJsonReified<RpcHistoryTransactionInfoResponse.Swap>(transaction.info.toString())
             ?: error("Parsing error: cannot parse json object")
+        val sourceTotal = info.from.amount.amount.toBigDecimalOrZero()
+        val sourceTotalInUsd = info.from.amount.usdAmount.toBigDecimalOrZero()
```

Comment:
info.from.amount -> info.from.amounts (there are multiple amounts in this field)